autogen.sh

```diff
@@ -39,8 +39,6 @@ if [ -z "$(which glib-gettextize)" ]; then
     exit 1
 fi
 
-test "${NO_NPM:-}" != "1" && tools/npm-install
```

Comment:
This actually helps me with cleaning up the -dist stuff as well. But this still feels incomplete - autogen.sh should ensure in *some way* that node_modules is available. Maybe with a simple `[ -e package-lock.json  ] || tools/node-modules`?

Reply:
I disagree with this.  The point here is that `node_modules` should be automatically created by make on an as-needed ("as" = "if and when") basis.  If we do a `WEBPACK_PACKAGES=''` build, for example, then `node_modules` will never be checked out.